src/client/Microsoft.Identity.Client/Http/HttpManager.cs

```diff
@@ -134,10 +135,30 @@ public async Task<HttpResponse> SendRequestAsync(
             logger.Warning("Request retry failed.");
             if (timeoutException != null)
             {
+                //If the correlation id is available, include it in the exception message
+                string msg = "Request to the endpoint timed out.";
+
+                if (headers != null && headers.Count > 0)
+                {
+                    var correlationId = headers[OAuth2Header.CorrelationId];
+                    string correlationIdMsg = headers.ContainsKey(OAuth2Header.CorrelationId) ?
```

Comment:
should we do a tryGet? and also check to make sure the id is not empty or null?

Reply:
The correlationId is always added to the headers so the test added is already testing that
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/07f9c3bb0dcd32a69e7eaaa4cef9dca58fdd0495/src/client/Microsoft.Identity.Client/OAuth2/OAuth2Client.cs#L228